ecocodejava.properties

```diff
@@ -1,13 +1,21 @@
 category=External Analysers
 description=Provide a list of static code analyzers (for Java language) to highlight code structures that may have a negative ecological impact: energy and resources over-consumption, "fatware", shortening terminals' lifespan, etc.
 homepageUrl=https://github.com/green-code-initiative/ecoCode
-publicVersions=1.2.1
+publicVersions=1.3.0
+archivedVersions=1.2.1
+
 
 defaults.mavenGroupId=io.ecocode
 defaults.mavenArtifactId=ecocode-java-plugin
 
 1.2.1.description=Initial Marketplace publication of plugin for SonarQube 9.9.+ LTS
-1.2.1.sqVersions=[9.9,LATEST]
+1.2.1.sqVersions=[9.9,9.9]
```

Comment:
The ranges of the versions you're archiving need to end with the current version (unless they're truly not compatible). Otherwise the Marketplace prompt users of e.g. SonarQube 10.1 who have 1.2.1 installed to upgrade to 1.3.0.

BTW, this is a semantic error, rather than one that would be caught by automated processes.

https://github.com/SonarSource/sonar-update-center-properties#moving-a-version-from-publicversions-to-archivedversions

Reply:
Thank you for your advise.
I wonder how to develop an automated test to detect this kind of problem (to avoid repeating the same error next time)... (cf. #426)

Reply:
It's a fair observation that some of these the-generation-job-will-break-if-you-do-this problems could / should be picked up automatically. And I have a hard-enough time getting actual bugs in this area fixed. :-/